### ISSUE TITLE
Mixing Linux and Windows containers

### ISSUE THREAD

**Issue by masoudr** [BretFisher/ama#99]
Hello,
I need your advice about using Windows and Linux containers simultaneously.
Currently, in my projects, I have both Windows and Linux containers that need to be run simultaneously and connect to each other. As you might know, there is an experimental feature in Docker desktop for that by selecting Windows containers and use `--platform` when running containers.
But it seems that this feature doesn't work properly and I get strange behavior when using this feature.
For example in one container I'm trying to build a Linux container with this Dockerfile:
```
FROM python:3.8
ENV PYTHONUNBUFFERED 1
RUN mkdir -p /app/build
COPY ./source /app
WORKDIR /app
ENTRYPOINT [ "python", "/app/build.py" ]
```
But when building it with `docker build --platform linux -t python .` I get this:
```
The command '/bin/sh -c mkdir -p /app/build' returned a non-zero code: 4294967295: failed to shutdown container: container c9be4... encountered an error during hcsshim::System::waitBackground: failure in a Windows system call: The virtual machine or container with the specified identifier is not running. (0xc0370110): subsequent terminate failed container c9be4... encountered an error during hcsshim::System::waitBackground: failure in a Windows system call: The virtual machine or container with the specified identifier is not running. (0xc0370110)
```
Which seems a bug related to the Windows container.
Long story short. I'm not sure what should I do now, I need to use Windows and Linux containers together. So one idea for me is to use Orchestrations like Swarm but this forces me to use another machine for my application which is not ideal for me. Can you suggest an alternative to my problem?
Thanks for your time.

**Comment by BretFisher** [comment]
I recommend posting that bug to the GH Issue repo for Windows DD https://github.com/docker/for-win/issues

Sorry, I don't know of an alternative other than using dedicated machines for Windows/Linux containers. I don't know if the new `docker buildx` commands work in the experimental platform feature.

**Comment by masoudr** [comment]
Thanks for the info.
I looked at `docker buildx` but it doesn't seem to build Linux/Windows containers. Anyway, I am trying to switch to orchestrations like a swarm. But I was wondering can I create a Linux VM with Hyper-V for my Linux containers and connect it to the Host which has Docker Desktop with Windows container. Do you think is it possible?

**Comment by BretFisher** [comment]
1. Yes a swarm can have different OS's and Architectures in it. They all run docker.
2. Docker Desktop is just for local testing and dev, not for making a multi-node swarm.
3. If you wanted a multi-node Swarm in Hyper-V, you'd create a new VM for each node (could be Windows or Linux)
4. I don't know if any of the file replication solutions will work across Linux+Windows, I've only seen them used on Linux.

**Comment by masoudr** [comment]
@BretFisher Thank you for the suggestion. I tried what you said and used a multi-node swarm.
It took me a long time to handle the shared persistent volume between Linux and Windows nodes. I found the [official docs](https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/persistent-storage#smb-mounts) and used the SMB mount. It is not a swarm aware tech but got the job done.
